graylog2-web-interface/src/views/logic/slices/undoRedoActions.tsx

```diff
@@ -18,39 +18,48 @@
 import type { AppDispatch } from 'stores/useAppDispatch';
 import type { RootState, ViewState } from 'views/types';
 import { selectRootUndoRedo, selectUndoRedoAvailability } from 'views/logic/slices/undoRedoSelectors';
-import { selectQuery, updateView } from 'views/logic/slices/viewSlice';
+import { isViewWidgetsEqualForSearch, selectQuery, updateView } from 'views/logic/slices/viewSlice';
 import { setCurrentRevision } from 'views/logic/slices/undoRedoSlice';
+import { selectRootView } from 'views/logic/slices/viewSelectors';
+import type View from 'views/logic/views/View';
+import isEqualForSearch from 'views/stores/isEqualForSearch';
+import type Search from 'views/logic/search/Search';
 
-const viewHandler = (state: ViewState, { hasToPushRevision, dispatch }: {
+const isViewSearchEqual = (first: Search, second: Search) => isEqualForSearch(first, second);
+const viewHandler = (state: ViewState, { hasToPushRevision, dispatch, currentView }: {
   hasToPushRevision: boolean,
-  dispatch: AppDispatch
-}): Promise<unknown> => dispatch(selectQuery(state.activeQuery)).then(() => dispatch(updateView(state.view, state.isDirty, { hasToPushRevision })));
+  dispatch: AppDispatch,
+  currentView: View,
+}): Promise<unknown> => dispatch(selectQuery(state.activeQuery)).then(() => {
+  const isDirty = !isViewWidgetsEqualForSearch(state.view, currentView) || !isViewSearchEqual(state.view.search, currentView.search);
```

Comment:
Maybe let's call this `shouldRecreateSearch` instead of `isDirty`? For me it was a bit confusing because views and the view slice already have a `dirty` attribute, which has a different meaning, since it also considers e.g. widget position updates.